### PR TITLE
Simplify AcousticBrainz extractor settings

### PR DESCRIPTION
**Pull request by zas**
- do not output error when this feature was never set
- display a tooltip on download button
- (always) set a translatable place holder for path input field
- let user edit the path directly
- ensure extractor validity is checked after browsing for it

# Summary

* This is a…
    * [x] Bug fix
    * [ ] Feature addition
    * [x] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-XXX

# Solution

# Action

### CODE REVIEW

**Review by phw — CHANGES_REQUESTED** [review]
We intentionally did not want to persist the found path if it is empty automatically. Instead empty path will trigger auto detection. I think this PR breaks this, because it will persist the detected path to settings.

We have setups were the path is rather dynamic and changes. E.g. for portable Windows version in changes on each start of Picard, for Snaps it changes on each app update. And even on macOS it changes when the user moves the app around.

Hence we should keep the use case of having the path empty. That is also consistent with how fpcalc is handled.

That also means `check_extractor_version` must still check for a version even if path is empty. But I agree we should not log errors in this case. Maybe if path is empty and no binary is found just log a info or warning.

There is also the "Use acoustic feature extraction" option we could check. If that's disabled we should definitely not try to detect the binary at all.

**Review by rdswift — COMMENTED** [review]
@phw has already mentioned all of the concerns that I had regarding use of the Windows Portable version.  The rest looks good to me.

**Comment by zas** [comment]
> We intentionally did not want to persist the found path if it is empty automatically.

Oh I missed this, I'll rework the PR.

So, when in "auto-detect" mode, we shouldn't save any path (and remove any existing path from config), right?

Can you confirm "auto-detect" should be triggered when:
- no path is set (=empty field), either one first run or because user cleared the existing path
- when invalid path is set (=something in the field but not a valid executable or invalid version)

On `save`, it reads the text from field, so I guess this is the reason a place holder was used instead.

So basically, place holder should be set to the path of the found path (if any), but no value should be set.

**Comment by zas** [comment]
I made changes so path is never saved if auto-detected.
Please test & review.

**Comment by rdswift** [comment]
I just checked the portable version, and it seems to work well.  :+1:

**Review by rdswift — COMMENTED** [review]
I just checked the portable version, and it seems to work well.  :+1:

**Comment by zas** [comment]
@phw please re-review, I addressed your comments (I think) and went a bit further in cleaning up the code. I detected few potential issues I tried to solve here.

**Review by phw — APPROVED** [review]
Works well now, good improvements
